pyam/utils.py

```diff
@@ -111,15 +111,20 @@ def write_sheet(writer, name, df, index=False):
             pass
 
 
-def read_pandas(path, default_sheet='data', *args, **kwargs):
+def read_pandas(path, sheet_name='data*', *args, **kwargs):
     """Read a file and return a pandas.DataFrame"""
     if isinstance(path, Path) and path.suffix == '.csv':
         df = pd.read_csv(path, *args, **kwargs)
     else:
         xl = pd.ExcelFile(path)
-        if len(xl.sheet_names) > 1 and 'sheet_name' not in kwargs:
-            kwargs['sheet_name'] = default_sheet
-        df = pd.read_excel(path, *args, **kwargs)
+        sheet_names = pd.Series(xl.sheet_names)
+        if len(sheet_names) > 1:
+            sheets = kwargs.pop('sheet_name', sheet_name)
+            # apply pattern-matching for sheet names (use * as wildcard)
+            sheets = sheet_names[pattern_match(sheet_names, values=sheets)]
+            df = pd.concat([xl.parse(s, *args, **kwargs) for s in sheets])
```

Comment:
This is pythonic....
in practice, particularly because dealing with many sheets generally means large data and longer load in time, I have done this with a for loop (actually looping pyam.IamDataFrame + pyam.append):
1. print each sheet name to give user feedback on the progress
2.  because if there is an error reading in a sheet, then this helps debug in which sheet a problem is located.
something like:
```
for s in sheets:
    print(f'loading {s}')
    df.append(pyam.IamDataFrame(xl, sheet_name=s),
                                                  inplace=True)
```
or as you have it, something like?
`df = df.append(xl.parse(s, *args, **kwargs))`

Reply:
One reason that I implemented it in this way is that `IamDataFrame.append()` is not very efficient - for every append, it checks whether there are any duplicate entries in the timeseries data. So first doing a concatenation in pandas and only then casting it to an IamDataFrame (and doing the validation for duplicates) should be much faster.

If you can run a test and feel that the additional logging is still helpful, maybe
```
lst = []
for s in sheets:
    logger.info(f'Reading sheet {s}')
    lst.append(xl.parse(s, *args, **kwargs))
df = pd.concat(lst)
```

Reply:
I suspected it was speed - and it is actually in the `pyam` checks where the good checks are done.
Anyways - great, I'll add this suggestion above.